src/routes/messages/messages.controller.spec.ts

```diff
@@ -425,4 +426,181 @@ describe('Messages controller', () => {
         });
     });
   });
+
+  describe('Get messages by Safe address', () => {
+    it('should group messages by date', async () => {
+      const chain = chainBuilder().build();
+      const safe = safeBuilder().build();
+      const messageCreationDate = faker.date.recent();
+      const messages = range(4).map(() =>
+        messageBuilder()
+          .with('safeAppId', null)
+          .with('created', messageCreationDate)
+          .build(),
+      );
+      const page = pageBuilder()
+        .with('previous', null)
+        .with('next', null)
+        .with('count', messages.length)
+        .with(
+          'results',
+          messages.map((m) => messageToJson(m)),
+        )
+        .build();
+      mockNetworkService.get.mockImplementation((url) => {
+        switch (url) {
+          case `${safeConfigUrl}/api/v1/chains/${chain.chainId}`:
+            return Promise.resolve({ data: chain });
+          case `${chain.transactionService}/api/v1/safes/${safe.address}`:
+            return Promise.resolve({ data: safe });
+          case `${chain.transactionService}/api/v1/safes/${safe.address}/messages/`:
+            return Promise.resolve({ data: page });
+          default:
+            return Promise.reject(`No matching rule for url: ${url}`);
+        }
+      });
+
+      await request(app.getHttpServer())
+        .get(`/v1/chains/${chain.chainId}/safes/${safe.address}/messages`)
+        .expect(200)
+        .expect(({ body }) => {
+          expect(body).toEqual(
+            pageBuilder()
+              .with('next', null)
+              .with('previous', null)
+              .with('count', messages.length)
+              .with('results', [
```

Comment:
nitpick: I like the way this test is written – it tries to go down to what is expected but not trying to match everything.

However maybe it's due to the size of the `results` but I think testing the entire schema returned is valuable. We can do that with a single message and then use partial-matching for bigger payloads.

Reply:
Yes I agree, I've added another test that checks the validity of a response consisting in a single `DateLabel` and `MessageItem` in [7293f5f](https://github.com/5afe/safe-client-gateway-nest/pull/279/commits/7293f5fdc5270406df511c6be56bcb0a5c38a728)